File: tools/GhidraDumpTypes.java
// Dump every defined data symbol's address, label, data type and size to CSV.
import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.Data;
import ghidra.program.model.listing.DataIterator;
import java.io.*;

public class GhidraDumpTypes extends GhidraScript {
    @Override
    public void run() throws Exception {
        String[] args = getScriptArgs();
        PrintWriter pw = new PrintWriter(new FileWriter(args[0]));
        pw.println("addr,label,type,size");
        DataIterator it = currentProgram.getListing().getDefinedData(true);
        int n = 0;
        while (it.hasNext()) {
            Data d = it.next();
            if (!d.getAddress().isMemoryAddress()) continue;
            String label = d.getLabel();
            String tn = d.getDataType().getName().replace(",", ";");
            pw.printf("0x%s,%s,%s,%d%n", d.getAddress(), label == null ? "" : label,
                      tn, d.getLength());
            n++;
        }
        pw.close();
        println("dumped " + n + " data items");
    }
}

File: tools/ghidra_dump_types.py
# Ghidra headless postScript (Jython): dump every defined data symbol's
# address, label, data type and size to CSV for the matching pipeline.
# Run: analyzeHeadless <proj> -process BRGlide.dll -noanalysis \
#        -postScript ghidra_dump_types.py <out_csv>
import sys
args = getScriptArgs()
out = open(args[0], 'w')
out.write('addr,label,type,size\n')
listing = currentProgram.getListing()
it = listing.getDefinedData(True)
n = 0
while it.hasNext():
    d = it.next()
    a = d.getAddress()
    if not a.isMemoryAddress():
        continue
    dt = d.getDataType()
    label = d.getLabel() or ''
    tn = dt.getName().replace(',', ';')
    out.write('0x%s,%s,%s,%d\n' % (a, label, tn, d.getLength()))
    n += 1
out.close()
print('dumped %d data items' % n)
